### DISCUSSION TITLE
스터디 최종 회고

### DISCUSSION THREAD

**Discussion by gw-lim** [code-itch/effective-typescript#68] [General]
## ✏️ 스터디 최종 회고
다음 내용들을 넣어 작성해주시면 감사하겠습니다.
- 소감
- 피드백

**Comment by qooktree1** [comment]
## 소감
- "이펙티브 타입스크립트" 책을 읽어보며 다시 보니 혼자 공부하기에 심오한 부분들이 있고 이해가 되지 않았던 부분들이 있었습니다. 스터디를 진행하며 다른 사람들이 책의 Item들에 대한 요약 및 그것에 대한 본인 견해를 들으며 점점 성장해나가는 저 자신을 바라볼 수 있었습니다. 비록 하나의 스터디이지만 다른 사람들이 Typescript, 프로젝트에서 사용하며 느꼈던 부분들, 더 나아가 가치관을 볼 수 있는 기회가 주어진 것 같아 스터디에 들어가기를 잘했다는 생각이 드네요..! 😋😋

## 피드백
- 일단, 스터디 전에 개인별로 최소 하나의 질문을 해야한다는 규칙을 정했는데 이 부분이 초반에 빨리 정해졌으면 하는 아쉬움이 있습니다.
  - 저에게 있어서 굉장히 효과도 좋았기 때문에 앞으로 스터디를 진행하게 된다면 이 부분을 참고할 것 같습니다. 😋

- 스터디를 진행하며 넘어가는 헷갈렸던 부분들이 조금씩 쌓이며 복습을 제대로 하지 못한 것 같습니다. 스터디 특성상 어쩔 수 없는 부분이라 생각이 들고 스터디가 끝난 후 다시 저를 복기하는 시간이 필요할 것 같습니다.

- 앞으로의 계획: 현재 제가 진행중에 있는 프로젝트 및 스터디들이 있기 때문에 앞으로의 스터디가 새로 생긴다면 참여를 못할 것 같아 너무나도 아쉽습니다..

**Comment by kses1010** [comment]
### 소감
- 이펙티브 타입스크립트를 혼자 공부하기에는 저만의 시야를 가지고 진행할 수도 있으나 스터디를 진행하면서 다양한 의견을 얻을 수 있었습니다. 또한 다같이 공부해야하는 의무가 생겨 혼자했더라면 끝까지 가지 못할뻔했으나 결국엔 완주했다는것이 가장 컸습니다.
- 스터디 장으로서 소감을 밝히자면 이 스터디를 잘 이끌었는지는 잘모르겠다는게 저의 소감입니다. 조금만 더 일찍 알았더라면 또는 더 적극적으로 참여도 가능하지 않을까에 대한 아쉬움도 있습니다. 그래도 그나마 만족스러운점은 초기 스터디원 그대로 끝까지 스터디를 마무리를 지었다는것입니다.
- 이펙티브 타입스크립트 스터디를 끝까지 같이 해주신 스터디원께 감사드립니다.

### 피드백
- 개인별로 최소 하나의 질문하기 규칙을 좀 더 빠르게 진행했어야 했는데에 대한 아쉬움이 있습니다.
- 이것을 깨닫기 위해 최소 2번의 스터디할 때마다 마지막으로 스터디 회고를 진행했어야했습니다. 만약에 다시 시작한다면 스터디 회고를 최소 한달에 한번정도 하는것을 고려해야겠습니다.
- 앞으로의 계획: 개발자가 되기위한 한달에 한권 읽기 스터디를 진행하고 싶은데 시간이 남는다면 의중을 들어보려고 합니다!

**Comment by Ssong-Q** [comment]
### 소감
- 이펙티브 타입스크립트 스터디를 통해 이펙티브 타입스크립트 책을 완독할 수 있었습니다. 모든 챕터를 정독한 것은 아니지만, 최소한 내가 맡은 부분에 대해서는 발표를 위해 내용 정리를 할 수 있었고, 또 모든 챕터를 정독하지 않았더라도, 다른 분들의 발표를 통해 다른 챕터들의 내용을 엿볼 수 있었습니다.
- 모든 스터디원들이 부트캠프의 과제, 강의와 프로젝트를 병행하며 스터디를 진행하느라 고생했다고 말씀드리고 싶습니다.

### 피드백
- 우선 내 자신에게 '초심을 잃지 않고 열심히 했냐'라고 묻는다면 그렇다라고 깔끔하게 대답을 못 할 것 같습니다. 중후반부부터는 바쁘다는 핑계로 다른 분들의 정리와 파트를 많이 살펴보지 못했기 때문입니다. 그래서 스터디는 끝났지만, 이 책 한 권을 결국 끝냈지만, 이 책을 정말 다 읽었냐?는 아닌 것 같습니다. 특히, 지금의 실력으로는 꽤 어려운 부분들과 이해하기 힘든 부분들이 있었기에 추후 실력을 더 쌓은 다음 한 번 더 복습차 봐야할 것 같습니다.
- 스터디 진행 방식은 꽤 마음에 들었고, 좋았다고 생각했어서, 다른 스터디를 진행하는데 있어서 많은 참고가 될 것 같습니다. 그동안 스터디장님께 고생 많으셨다고 말씀해드리고 싶습니다.
- 앞으로의 계획: 현재 이력서 수정, 구직, 다른 스터디들로 바쁜 와중이라 또 다른 내용으로 스터디를 이어나갈 수 있을지는 모르겠습니다.

**Comment by BeMatthewsong** [comment]
## 소감
- 타입스크립트를 접한 지 얼마 안 되었을 때 이 스터디를 시작했던 게 생각납니다. 그때와 다르게 저는 타입스크립트를 봤을 때 두려움이 많이 없어지고, 타입스크립트에 대한 식견이 많이 넓혀졌다는 게 느껴졌습니다. 물론 완전히 다 안다는 건 아니지만 작년 말의 저와는 많이 다름을 느꼈던 것 같습니다.
- 이 스터디는 스터디장이신 상희님이 되게 체계적으로 구성해주시고 열심히 참여해주신 스터디원들 덕분에 타입스크립트 이외에도 코드를 대하는 태도와 애정을 많이 배울 수 있었습니다. 
- 기술적인 내용을 발표를 통해 입밖으로 꺼낼 수 있는 기회를 많이 가져서 기술적인 이야기를 나눠도 예전보다 어색함을 덜어낸 것 같아 좋습니다.

## 피드백
- 확실히 프로젝트랑 병행하면서 해서 점점 팀원 파트를 소홀히 대한 점이 아쉽습니다. 그래서 중간에 스터디장이신 상희님이 중간 회고를 통해 질문 1개 꼭 만들어오게 해서 다른 팀원들의 파트를 눈여겨볼 수 있게 해서 좋았습니다.
- 이펙티브 타입스크립트는 제 수준에서 완전히 이해할 수 있는 책은 아니었지만, 여러 번 이 책을 곱씹다보면 타입스크립트를 더 사랑하거나 더 증오할 것 같습니다 ㅋㅋ

## 앞으로의 계획 
앞으로 저는 다른 스터디와 개인적으로 해나가야 하는 학습과 프로젝트가 있어서 이 멤버들과의 스터디는 진행하기 어려울 것 같습니다.
대신에 이 오가니제이션을 통해 취업하더라도 정보를 공유할 수 있는 공간으로 남겼으면 하는 바람이 있습니다. 다들 동의하신다면 다 같이 노력해보는 건 어떠신가요?

**Comment by gw-lim** [comment]
## 소감
- 혼자라면 끝까지 완독하게 어려웠을 이펙티브 타입스크립트를 스터디원분들 덕분에 끝까지 완주할 수 있었습니다. 혼자서만 읽었다면 이해가 되지 않는 부분들을 제대로 찾아보지 않고 넘어갔을 것 같지만, 다른 분들에게 소개해야한다는 부분 덕분에 저 또한 자세히 내용을 찾아보고 남들에게 설명할 수 있을 정도까지 이해할 수 있도록 노력했던 것 같습니다. 다른 분들도 노력해주시고 추가적인 내용까지 찾아와주시는 모습을 보면서 정말 대단하다고 생각했고, 감사하다는 말씀 전하고 싶습니다.
- 타입스크립트를 처음 배우기 시작하면서 해당 스터디를 진행하게 되었는데, 처음에 비해 타입스크립트와 훨씬 더 친해졌다고 느끼고 있습니다. 특히 다른 분들의 의견을 같이 공유하면서 타입스크립트에 대한 지식 뿐만 아니라 개발에 대한 생각과 시각을 넓히는 경험이 된 것 같습니다.

## 피드백
- 초반에는 다른 분들의 파트까지 모두 읽자는 마음이 있었는데, 확실히 프로젝트와 병행하다보니 시간이 부족해져 제 파트만 읽게 되었습니다. 이로 인해 질문을 적극적으로 못하게 된 부분이 아쉽게 느껴집니다. 도중에 정해진 최소 1개 질문 원칙이 더 일찍 정해졌으면 좋았을 것 같다고 생각합니다.
- 이번에 진행한 스터디가 앞으로 제가 생각하는 이상적인 스터디가 될 것 같습니다. 다른 스터디를 진행한다면 이번 경험을 토대로 많이 참고하여 진행하고 싶습니다.
- 앞으로 저는 개인적으로 진행하고 싶은 프로젝트가 있어서 추가적인 스터디를 하기는 어려울 것 같아 너무 아쉽습니다..
